sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -208,6 +208,7 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]]
     }
 
     def recursiveTransform(arg: Any): AnyRef = arg match {
+      case ue: UnpivotExpr => ue.withNewChildren(ue.exprs.map(transformExpression))
```

Comment:
This is a very special treatment in a very generic method. Not nice, but the best way I could find to make `mapExpressions` tranform `UnpivotExpr.exprs` as a top-level expression. See comment on top-level expression in PR discussion.